sns_ik_examples/src/ik_tests.cpp

```diff
@@ -329,57 +329,57 @@ void test(ros::NodeHandle& nh, double num_samples_pos, double num_samples_vel,
   ROS_INFO_STREAM("KDL nullspace success rate: "
                    << 100.0 * kdlPos_ns_successRate << ", avg l2_norm ratio: "
                    << kdlPos_ns_avgL2Score << ", avg time: " << kdlPos_ns_avgTime);
-  total_time=0;
-  success=0;
-  std::vector<double> tracPos_indivTime;
-
-  ROS_INFO_STREAM("*** Testing TRAC-IK with "<<num_samples_pos<<" random samples");
-  if (use_nullspace_bias_task)
-    ROS_WARN("TRAC-IK does not support secondary tasks.");
+  #if USE_TRAC
+    total_time=0;
+    success=0;
+    std::vector<double> tracPos_indivTime;
+    TRAC_IK::TRAC_IK tracik_solver(chain_start, chain_end, urdf_param, timeout, eps);
+    valid = tracik_solver.getKDLChain(chain);
+    if (!valid) {
+      ROS_ERROR("There was no valid KDL chain found");
+      return;
+    }
+    valid = tracik_solver.getKDLLimits(ll,ul);
+    if (!valid) {
+      ROS_ERROR("There were no valid KDL joint limits found");
+      return;
+    }
+    assert(chain.getNrOfJoints() == ll.data.size());
+    assert(chain.getNrOfJoints() == ul.data.size());
+    ROS_INFO ("TRAC-IK Using %d joints",chain.getNrOfJoints());
```

Comment:
You already have a copy of the chain and joint limits from the SNS IK solver. Shouldn't need to overwrite them via the KDL solver since it should return the same values.